packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-native-deps.js

```diff
@@ -0,0 +1,123 @@
+                if (!firstRun) {
+                    nativeDependencies.length = 0;
+                }
+                firstRun = false;
+            }
+        },
+        async writeBundle() {
+            if (shouldCopyNodeModules) {
+                await Promise.all(
+                    nativeDependencies.map(async dependency => {
+                        await copyJsModule(dependency.dir, join(nodeModulesPath, dependency.name));
+                        for (const transitiveDependency of await getTransitiveDependencies(
+                            dependency.name,
+                            externals
+                        )) {
+                            await copyJsModule(
+                                dirname(require.resolve(`${transitiveDependency}/package.json`)),
+                                join(nodeModulesPath, dependency.name, "node_modules", transitiveDependency)
+                            );
+                        }
+                    })
+                );
+                await writeNativeDependenciesJson({ nativeDependencies, outputDir, widgetName });
+            }
+        }
+    };
+}
+
+async function writeNativeDependenciesJson({ nativeDependencies, outputDir, widgetName }) {
+    if (nativeDependencies.length === 0) {
+        return;
+    }
+    const dependencies = {};
+    for (const dependency of nativeDependencies) {
+        dependencies[dependency.name] = (await readJson(join(dependency.dir, "package.json"))).version;
+    }
+    await writeJson(join(outputDir, `${widgetName}.json`), { nativeDependencies: dependencies }, { spaces: 2 });
+}
+
+async function hasNativeCode(dir) {
+    const packageContent = await promises.readdir(dir, { withFileTypes: true });
+
+    if (packageContent.some(file => /^(ios|android|.*\.podspec)$/i.test(file.name))) {
+        return true;
+    }
+
+    for (const file of packageContent) {
+        if (file.isDirectory() && (await hasNativeCode(join(dir, file.name)))) {
+            return true;
+        }
+    }
+    return false;
+}
+
+async function getTransitiveDependencies(packageName, externals) {
+    const queue = [packageName];
+    const result = new Set();
+    while (queue.length) {
+        const next = queue.shift();
+        if (result.has(next)) {
+            continue;
+        }
+        const isExternal = externals.some(external =>
+            external instanceof RegExp ? external.test(next) : external === next
+        );
+        if (isExternal) {
+            continue;
+        }
+
+        if (next !== packageName) {
+            result.add(next);
+        }
+        const packageJson = await readJson(require.resolve(`${next}/package.json`));
+        queue.push(...Object.keys(packageJson.dependencies || {}));
+    }
+    return Array.from(result);
+}
+
+async function copyJsModule(from, to) {
+    await copy(from, to, {
+        filter: async path =>
+            (await promises.lstat(path)).isDirectory()
```

Comment:
🎄 can we maybe try to express it with a single regexp and/or glob?

Reply:
Improved a bit the regex to identify relevant files and exclude folders

Reply:
Still feels like you are doing a glob search

Reply:
We dded a glob copy using recursive-copy lib